### PR TITLE
feat: Define ServiceLoadBalancer Configuration API

### PR DESCRIPTION
**Pull request by dlipovetsky**
**What problem does this PR solve?**:

**Which issue(s) this PR fixes**:
Fixes #

**How Has This Been Tested?**:

**Special notes for your reviewer**:

### CODE REVIEW

File: api/v1alpha1/addon_types.go [hunk 1]
```diff
@@ -257,4 +257,28 @@ type ServiceLoadBalancer struct {
 	// +kubebuilder:validation:Enum=MetalLB
 	// +kubebuilder:validation:Required
 	Provider string `json:"provider"`
+
+	// Configuration for the chosen ServiceLoadBalancer provider.
+	// +kubebuilder:validation:Optional
+	Configuration *ServiceLoadBalancerConfiguration `json:"configuration"`
```
Comment (dkoshkin) [inline api/v1alpha1/addon_types.go:263]:
```suggestion
	Configuration *ServiceLoadBalancerConfiguration `json:"configuration,omitempty"`
```

File: api/v1alpha1/addon_types.go [hunk 2]
```diff
@@ -257,4 +257,28 @@ type ServiceLoadBalancer struct {
...
+}
+
+type ServiceLoadBalancerConfiguration struct {
+	// AddressRanges is a list of IPv4 or IPv6 address ranges from which
+	// the provider can choose a load balancer address.
+	AddressRanges []AddressRange `json:"addressRanges"`
```
Comment (dkoshkin) [inline api/v1alpha1/addon_types.go:269]: Should these be optional?
```suggestion
	// AddressRanges is a list of IPv4 or IPv6 address ranges from which
	// the provider can choose a load balancer address.
	// +kubebuilder:validation:Optional
	AddressRanges []AddressRange `json:"addressRanges,omitempty"`
```

File: api/v1alpha1/addon_types.go (hunk 1 above)
Comment (dlipovetsky) [reply api/v1alpha1/addon_types.go:263]: Thanks, will fix!

File: api/v1alpha1/addon_types.go (hunk 2 above)
Comment (dlipovetsky) [reply api/v1alpha1/addon_types.go:269]: Nope, they're required. Will fix.

**Review by dkoshkin — APPROVED** [review]

File: api/v1alpha1/addon_types.go [hunk 3]
```diff
@@ -257,4 +257,29 @@ type ServiceLoadBalancer struct {
...
+
+	// Configuration for the chosen ServiceLoadBalancer provider.
+	// +kubebuilder:validation:Optional
+	Configuration *ServiceLoadBalancerConfiguration `json:"configuration,omitempty"`
+}
+
+type ServiceLoadBalancerConfiguration struct {
+	// AddressRanges is a list of IPv4 or IPv6 address ranges from which
+	// the provider can choose a load balancer address.
+	// +kubebuilder:validation:MinItems=1
```
Comment (jimmidyson) [inline api/v1alpha1/addon_types.go:269]: Please add  required too - I've been advised to do this upstream even though it is a default as there have been reports of this not being consistently handled in controller-gen.
```suggestion
	// +kubebuilder:validation:Required
	// +kubebuilder:validation:MinItems=1
```

Comment (dlipovetsky) [reply api/v1alpha1/addon_types.go:269]: Done in #778

**Comment by dlipovetsky** [comment]
Replaced by #778 because this is from a public fork, and can't run some of the required checks.

**Pull request closed without merging**